plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.elasticsearch.decoders.DoubleDecoder;
+import io.trino.plugin.elasticsearch.decoders.IntegerDecoder;
+import io.trino.plugin.elasticsearch.decoders.VarcharDecoder;
+import io.trino.spi.predicate.Domain;
+import io.trino.spi.predicate.Range;
+import io.trino.spi.predicate.TupleDomain;
+import io.trino.spi.predicate.ValueSet;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.plugin.elasticsearch.ElasticsearchQueryBuilder.buildSearchQuery;
+import static io.trino.spi.type.DoubleType.DOUBLE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static org.testng.Assert.assertEquals;
+
+public class TestElasticsearchQueryBuilder
+{
+    static ElasticsearchColumnHandle name = new ElasticsearchColumnHandle("name", VARCHAR, new VarcharDecoder.Descriptor("name"), true);
+    static ElasticsearchColumnHandle age = new ElasticsearchColumnHandle("age", INTEGER, new IntegerDecoder.Descriptor("age"), true);
+    static ElasticsearchColumnHandle score = new ElasticsearchColumnHandle("score", DOUBLE, new DoubleDecoder.Descriptor("score"), true);
+
+    @Test
+    public void testMatchAll()
+    {
+        QueryBuilder res = buildSearchQuery(TupleDomain.none(), Optional.empty());
+        assertQueryBuilder(res, new MatchAllQueryBuilder());
+    }
+
+    @Test
+    public void testOneConstraint()
+    {
+        Map<ElasticsearchColumnHandle, Domain> domains = ImmutableMap.<ElasticsearchColumnHandle, Domain>builder()
+                .put(age, Domain.singleValue(INTEGER, 1L))
+                .build();
+
+        QueryBuilder query = new BoolQueryBuilder().filter(new TermQueryBuilder(age.getName(), 1));
+
+        QueryBuilder res = buildSearchQuery(TupleDomain.withColumnDomains(domains), Optional.empty());
+        assertQueryBuilder(res, query);
```

Comment:
Thanks. I hava rewrote those unit tests, use `assertQueryBuilder`  function to  input `domains` and assert with `expected QueryBuilder`. 
I think it would be clearer to read than before.